lib/internal/bootstrap/node.js

```diff
@@ -319,6 +319,18 @@ ObjectDefineProperty(process, 'features', {
 }
 
 const { emitWarning, emitWarningSync } = require('internal/process/warning');
+const { getOptionValue } = require('internal/options');
+
+let hasTypeScriptSupport;
+ObjectDefineProperty(process.features, 'typescript', {
+  __proto__: null,
+  get() {
+    return hasTypeScriptSupport ??= !!getOptionValue('--experimental-strip-types');
```

Comment:
This can be quickly checked on the userland. Why should we add this to the core?

Reply:
As explained in https://github.com/nodejs/node/issues/54294, it's not trivial to check. I think at least `process.env.NODE_OPTIONS` and `process.execArgv` need to be considered and I'm not sure if there aren't more edge cases.